### DISCUSSION TITLE
Pin.DRIVE_2

### DISCUSSION THREAD

**Discussion by sk8board** [micropython/micropython#16067] [RP2040 / Pico]
The RP2040 Datasheet shows a GPIO pin can be set for different drive current (2mA, 4mA, 8mA, and 12mA).

![image](https://github.com/user-attachments/assets/79785f67-d554-4cf4-8b0f-f752af8061b8)

The MicroPython documentation shows `machine.Pin.DRIVE_x` is dependent upon the port.

https://docs.micropython.org/en/latest/library/machine.Pin.html#machine-pin

I noticed the following code received the following error.

```py
from machine import Pin

a = Pin(0, Pin.OUT, Pin.DRIVE_2)

a.value(1)

# AttributeError: type object 'Pin' has no attribute 'DRIVE_2'
```

Does this mean the `DRIVE_x` attribute is not implemented in the rp2 port?

If so, what is the default DRIVE setting for the rp2 port?

**Comment by fmafma** [comment]
I don't see anything in the code (https://github.com/micropython/micropython/blob/master/ports/rp2/machine_pin.c), so the default value should apply: 0x01 -> 4mA

**Comment by fmafma** [comment]
But I think you can change it after you created the Pin, by writing in the matching register, using `machine.mem32[]` mecanism.
